### DISCUSSION TITLE
Page returns 404 before being redirected in Cloudflare Pages SSG

### DISCUSSION THREAD

**Discussion by rayblair06** [nuxt-modules/i18n#2970] [Q&A]
I'm not sure if this is a bug or a misconfiguration on our side, but I'm trying to understand how locale redirects are supposed to work with i18n in terms of SSG. We're using Nuxt 3, hosted on Cloudflare Pages, and we pre-render our pages with the locale prefix, so pages are organized as `/en/foo/index.html`, `/es/foo/index.html`, etc.

However, pages that don't match the file routing currently return a `404` status code before being redirected to the correct filepath.

For example, with the `prefix` strategy, accessing `/foo` initially results in a `404` error before redirecting to `/en/foo`. This likely occurs because `/foo.html` doesn't exist; the actual file is `/en/foo/index.html` via Cloudflare Pages. While i18n on the client side handles the redirection correctly, returning a `404` status code on the initial request can negatively impact our SEO.

We use dynamic urls so manually adding redirects isn't really an option.

Note: We experienced this issue with the `prefix_except_default` strategy as well, so it isn't exclusive to any specific routing strategy within i18n.

**Comment by kamding957** [comment]
Is your problem solved? I have a similar problem.

**Reply by rayblair06** [reply]
It’s been a while since I worked on the project, but if I remember correctly, we implemented a function to dynamically create redirects for prefixed routes, such as redirecting `/en/foo` to `/foo`, using Nuxt 3's route rules (https://nuxt.com/docs/guide/concepts/rendering#route-rules). These rules were passed to Cloudflare Pages during the build process (https://developers.cloudflare.com/pages/configuration/redirects/). Fortunately, the URLs were unique to each language, so clashes were unlikely.
